tree/ntuple/v7/src/RField.cxx

```diff
@@ -504,6 +505,17 @@ void ROOT::Experimental::Detail::RFieldBase::ConnectPageSink(RPageSink &pageSink
       }
       SetColumnRepresentative(rep);
    }
+   if (pageSink.GetWriteOptions().GetHasSmallClusters()) {
+      ColumnRepresentation_t rep = GetColumnRepresentative();
+      for (auto &colType : rep) {
+         switch (colType) {
+         case EColumnType::kSplitIndex64: colType = EColumnType::kSplitIndex32; break;
+         case EColumnType::kIndex64: colType = EColumnType::kIndex32; break;
+         default: break;
+         }
+      }
```

Comment:
Can't we squash this somehow (also above)?  I'm worried about duplicated/hard-to-maintain code if the number of fixups here keeps growing.

Reply:
I factored the column adjustments it out in its own method.